### ISSUE TITLE
Attribution changed delegate being called on Android, but not iOS.

### ISSUE THREAD

**Issue by NathanWarden** [adjust/unity_sdk#210]
Hello, I might be doing something wrong, but the attribution changed delegate isn't being called in our games on iOS and haven't seen this in subsequent runs either. When checking the attribution itself via AdjustiOS.GetAttribution() it gets the value pretty quickly during the first run and instantly during subsequent runs. It gets called as expected on Android. Everything else from Adjust seems to be working on both platforms. Does Adjust recognize attribution as being changed when it first gets the attribution? Thanks!

So, in summary: Both iOS and Android get attribution values, but Android calls the attribution changed delegate and iOS does not, so I have to poll for the value on iOS.

Thanks!

**Comment by uerceg** [comment]
Hi @NathanWarden 

Are you using **Adjust** game object we provide as part of the plugin? Or are you having custom game object of your own where you are using **Adjust.cs** script? Each method of **AdjustConfig** class used to set callbacks has also one additional parameter (we call it **sceneName**) and which defaults to **"Adjust"** value. This parameter represents the name of game object where **Adjust.cs** script is loaded with its methods we are trying to call from native code. In case you're using you custom game object, try to pass its name as 2nd parameter when setting the callback methods and let us know if iOS callbacks are working as expected or are you still facing some issues.

Cheers

**Comment by NathanWarden** [comment]
@uerceg Thanks, I'll try that as it might be the issue!

**Comment by pfay-fc** [comment]
Note that this can also happen if you don't have any adjust prefab in your scene at all. If you manually initialize adjust, most other things will work, but you need the prefab for these iOS callbacks.

**Comment by uerceg** [comment]
@pfay-fc 

Thanks for the comment. In theory, scenario you described doesn't really need prefab for iOS callbacks to work. Yes, in case you leave default parameter in callback setter methods to have default `Adjust` value because prefab is pretty much introducing game object named Adjust in the scene, so it will make things work automagically. But you can still chose not to add prefab to the scene and pass as second parameter name of your game object which handles `Adjust.cs` script.

In case you guys still have some issues / questions, feel free to comment/reopen.

Cheers
